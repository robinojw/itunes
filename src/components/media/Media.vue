<template>
  <a
    :href="result.collectionViewUrl"
    target="_blank"
    rel="noopener noreferrer"
    class="media"
  >
    <picture>
      <source :srcset="result.artworkUrl100" type="image/jpeg" />
      <img
        :src="result.artworkUrl100"
        :alt="result.artistName"
        loading="lazy"
      />
    </picture>
    <p class="collection-name">{{ result.collectionName }}</p>
    <p class="artist">{{ result.artistName }}</p>
  </a>
</template>

<script lang="ts">
import './media.scss';
import { Vue } from 'vue-property-decorator';
export default Vue.extend({
  props: {
    result: {}
  }
});
</script>
