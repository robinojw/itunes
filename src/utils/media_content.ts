export interface MediaContent {
  data: Message;
}

export interface Message {
  message: SearchResults;
}

export interface SearchResults {
  resultCount: number;
  results: Array<Result>;
}

export interface Result {
  wrapperType: string;
  kind: string;
  artistId: number;
  collectionId: number;
  trackId: number;
  artistName: string;
  collectionName: string;
  collectionType?: string | null;
  amgArtistId?: number | null;
  copyright: string | null;
  trackName: string;
  collectionCensoredName: string;
  trackCensoredName: string;
  artistViewUrl: string;
  collectionViewUrl: string;
  feedUrl: string;
  trackViewUrl: string;
  artworkUrl30: string;
  artworkUrl60: string;
  artworkUrl100: string;
  collectionPrice: number;
  trackPrice: number;
  trackRentalPrice: number;
  collectionHdPrice: number;
  trackHdPrice: number;
  trackHdRentalPrice: number;
  releaseDate: string;
  collectionExplicitness: string;
  trackExplicitness: string;
  trackCount: number;
  country: string;
  currency: string;
  primaryGenreName: string;
  contentAdvisoryRating: string;
  artworkUrl600: string;
  genreIds?: string[] | null;
  genres?: string[] | null;
}
