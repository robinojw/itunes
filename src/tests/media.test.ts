import 'jest';
require('jsdom-global')();
import { mount } from '@vue/test-utils';
import Media from '../components/media/Media.vue';
import { Result } from '@/utils/media_content';

const MOCK_RESULT = {} as Result;
MOCK_RESULT.artistName = 'Oasis';
MOCK_RESULT.artworkUrl100 =
  'https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg';
MOCK_RESULT.collectionName = "(What's the Story) Morning Glory?";
MOCK_RESULT.collectionViewUrl =
  'https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg';

describe('Mounted App', () => {
  const wrapper = mount(Media, {
    propsData: {
      result: MOCK_RESULT
    }
  });

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it('renders media component correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders the correct artist name', () => {
    expect(wrapper.html()).toContain(
      `<p class="artist">${MOCK_RESULT.artistName}</p>`
    );
  });

  it('renders the correct collection name', () => {
    expect(wrapper.html()).toContain(
      `<p class="collection-name">${MOCK_RESULT.collectionName}</p>`
    );
  });
});
