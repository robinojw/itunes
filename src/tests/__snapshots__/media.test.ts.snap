// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<a
  class="media"
  href="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
  rel="noopener noreferrer"
  target="_blank"
>
  <picture>
    <source
      srcset="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
      type="image/jpeg"
    />
     
    <img
      alt="Oasis"
      loading="lazy"
      src="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
    />
  </picture>
   
  <p
    class="collection-name"
  >
    (What's the Story) Morning Glory?
  </p>
   
  <p
    class="artist"
  >
    Oasis
  </p>
</a>
`;

exports[`Mounted App renders media component correctly 1`] = `
<a
  class="media"
  href="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
  rel="noopener noreferrer"
  target="_blank"
>
  <picture>
    <source
      srcset="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
      type="image/jpeg"
    />
     
    <img
      alt="Oasis"
      loading="lazy"
      src="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
    />
  </picture>
   
  <p
    class="collection-name"
  >
    (What's the Story) Morning Glory?
  </p>
   
  <p
    class="artist"
  >
    Oasis
  </p>
</a>
`;

exports[`Mounted Media Component 1`] = `
<a
  class="media"
  href="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
  rel="noopener noreferrer"
  target="_blank"
>
  <picture>
    <source
      srcset="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
      type="image/jpeg"
    />
     
    <img
      alt="Oasis"
      loading="lazy"
      src="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
    />
  </picture>
   
  <p
    class="collection-name"
  >
    (What's the Story) Morning Glory?
  </p>
   
  <p
    class="artist"
  >
    Oasis
  </p>
</a>
`;

exports[`Mounted Media renders media component correctly 1`] = `
<a
  class="media"
  href="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
  rel="noopener noreferrer"
  target="_blank"
>
  <picture>
    <source
      srcset="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
      type="image/jpeg"
    />
     
    <img
      alt="Oasis"
      loading="lazy"
      src="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
    />
  </picture>
   
  <p
    class="collection-name"
  >
    (What's the Story) Morning Glory?
  </p>
   
  <p
    class="artist"
  >
    Oasis
  </p>
</a>
`;

exports[`renders media component correctly 1`] = `
<a
  class="media"
  href="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
  rel="noopener noreferrer"
  target="_blank"
>
  <picture>
    <source
      srcset="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
      type="image/jpeg"
    />
     
    <img
      alt="Oasis"
      loading="lazy"
      src="https://is2-ssl.mzstatic.com/image/thumb/Music113/v4/f1/91/ef/f191effe-4cf2-932b-f2cf-a724454d2d49/source/100x100bb.jpg"
    />
  </picture>
   
  <p
    class="collection-name"
  >
    (What's the Story) Morning Glory?
  </p>
   
  <p
    class="artist"
  >
    Oasis
  </p>
</a>
`;
